components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2378,6 +2416,206 @@ private static JWT encryptWithRSA(JWTClaimsSet jwtClaimsSet, JWEAlgorithm encryp
     }
 
     /**
+     * Encrypt JWT id token using RSA algorithm.
+     *
+     * @param signedJwt           contains signed JWT body
+     * @param encryptionAlgorithm JWT signing algorithm
+     * @param spTenantDomain      Service provider tenant domain
+     * @param clientId            ID of the client
+     * @return encrypted JWT token
+     * @throws IdentityOAuth2Exception
+     */
+    private static JWT encryptWithRSA(SignedJWT signedJwt, JWEAlgorithm encryptionAlgorithm,
+                                      EncryptionMethod encryptionMethod, String spTenantDomain, String clientId)
+            throws IdentityOAuth2Exception {
+
+        try {
+            if (StringUtils.isBlank(spTenantDomain)) {
+                spTenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+                if (log.isDebugEnabled()) {
+                    log.debug("Assigned super tenant domain as signing domain when encrypting id token for " +
+                            "client_id: " + clientId);
+                }
+            }
+            String jwksUri = getSPJwksUrl(clientId, spTenantDomain);
+
+            if (StringUtils.isBlank(jwksUri)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Jwks uri is not configured for the service provider associated with " +
+                            "client_id: %s. Checking for x509 certificate", clientId));
+                }
+                return encryptUsingSPX509Certificate(signedJwt, encryptionAlgorithm, encryptionMethod, spTenantDomain,
```

Comment:
Since we don't have JWKS URI configured, we are attempting to encrypt using the x509 cert. 
What happens if the app doesn't have a x509 cert configured?

Have we handled that in a encryptUsingSPX509Certificate() method?

Reply:
IdentityOAuth2Exception is thrown by getX509CertOfOAuthApp(clientId, spTenantDomain) method , when certificate is not configured.

Added a null check as below to handle when public key is null scenarios,

```
if (publicKey == null) {
            throw new IdentityOAuth2Exception("Error while retrieving public key from X509 cert of oauth app with "
                   + "client_id: " + clientId + " of tenantDomain: " + spTenantDomain);
} 
```

Reply:
IdentityOAuth2Exception is thrown by getX509CertOfOAuthApp(clientId, spTenantDomain) method , when certificate is not configured.

Added a null check as below to handle when public key is null scenarios,

```
if (publicKey == null) {
            throw new IdentityOAuth2Exception("Error while retrieving public key from X509 cert of oauth app with "
                   + "client_id: " + clientId + " of tenantDomain: " + spTenantDomain);
} 
```